### PR TITLE
[v6r19] Fix TS too clever method + formatting

### PR DESCRIPTION
**Pull request by phicharp**
Sorry as I had prepared some formatting changes, they come along, not a big deal I hope...

BEGINRELEASENOTES
* .pylintrc
CHANGE: disable redefined-variable-type
CHANGE: max-nested-blocks=10 due to the many tests of result['OK']

* TransformationSystem
FIX: Create tasks with as many targets as possible in groupByReplicas

* All
CHANGE: use autopep8 for auto-formatting with following exceptions:
   * tabs = 2 spaces and not 4
   * line length check disabled (i.e. 120 characters instead of 80)
   * Option for autopep8 are: --ignore E111,E101,E501

ENDRELEASENOTES

### CODE REVIEW

File: TransformationSystem/Client/Utilities.py [hunk 1]
```diff
@@ -140,7 +147,7 @@ def groupByReplicas( self, files, status ):
           tasksLfns = breakListIntoChunks( lfns, self.groupSize )
           lfnsInTasks = []
           for taskLfns in tasksLfns:
-            if ( flush and not groupSE ) or ( len( taskLfns ) >= self.groupSize ):
+            if flush or ( len( taskLfns ) >= self.groupSize ):
```
Comment (phicharp) [inline TransformationSystem/Client/Utilities.py]: This is the real change in the TS logic

File: TransformationSystem/Client/Utilities.py [hunk 2]
```diff
@@ -115,7 +122,7 @@ def groupByReplicas( self, files, status ):
     tasks = []
     nTasks = 0
 
-    if not len( files ):
+    if files:
```
Comment (andresailer) [inline TransformationSystem/Client/Utilities.py]: `if not files`

File: DataManagementSystem/Client/DataManager.py [hunk 3]
```diff
@@ -37,7 +37,8 @@
 __RCSID__ = "$Id$"
 
 def _isOlderThan( stringTime, days ):
-  timeDelta = timedelta( days = days )
+  """ Check if a time stamp is older than a gieven number of days """
```
Comment (andresailer) [inline DataManagementSystem/Client/DataManager.py]: given

File: TransformationSystem/Client/Utilities.py (hunk 2 above)
Comment (andresailer) [reply TransformationSystem/Client/Utilities.py]: this was even caught by the tests.

Comment (phicharp) [reply TransformationSystem/Client/Utilities.py]: Oups, well caught!

File: .pylintrc [hunk 4]
```diff
@@ -16,14 +16,15 @@ persistent=yes
...
+disable=multiple_types
 
 
 [MESSAGES CONTROL]
 
 # Enable the message, report, category or checker with the given id(s). You can
 # either give multiple identifier separated by comma (,) or put this option
 # multiple time.
-#enable=
+enable=C
```
Comment (fstagni) [inline .pylintrc]: Why?

File: .pylintrc [hunk 5]
```diff
@@ -16,14 +16,15 @@ persistent=yes
 # List of plugins (as comma separated values of python modules names) to load,
 # usually to register additional checkers.
 load-plugins=
+disable=multiple_types
```
Comment (fstagni) [inline .pylintrc]: There's already a "disable" at line 37. And this specific one has no effect, as "multiple_types" is not a pylint message:

```
(DIRAC)  fstagni@DeskTof  ~/pyDevs/DIRAC   PilotSynchronizer  pylint --list-msgs | grep multiple_types
(DIRAC)  ✘ fstagni@DeskTof  ~/pyDevs/DIRAC   PilotSynchronizer  
```
What you probably look for is:
```
(DIRAC)  ✘ fstagni@DeskTof  ~/pyDevs/DIRAC   PilotSynchronizer  pylint --list-msgs | grep redefined-variable-type
:redefined-variable-type (R0204): *Redefinition of %s type from %s to %s*
```
which is already inserted in the integration branch (you can backport here if you want)

File: DataManagementSystem/Client/DataManager.py [hunk 6]
```diff
@@ -88,18 +89,18 @@ def __init__( self, catalogs = None, masterCatalogOnly = False, vo = False ):
...
-    self.fc = FileCatalog( catalogs = catalogsToUse, vo = self.vo )
+    self.fileCatalog = FileCatalog( catalogs=catalogsToUse, vo=self.voName )
     self.accountingClient = None
     self.resourceStatus = ResourceStatus()
-    self.ignoreMissingInFC = Operations( self.vo ).getValue( 'DataManagement/IgnoreMissingInFC', False )
-    self.useCatalogPFN = Operations( self.vo ).getValue( 'DataManagement/UseCatalogPFN', True )
-    self.dmsHelper = DMSHelpers( vo = vo )
+    self.ignoreMissingInFC = Operations( self.voName ).getValue( 'DataManagement/IgnoreMissingInFC', False )
+    self.useCatalogPFN = Operations( self.voName ).getValue( 'DataManagement/UseCatalogPFN', True )
+    self.dmsHelper = DMSHelpers( vo=vo )
```
Comment (fstagni) [inline DataManagementSystem/Client/DataManager.py]: `vo=self.voName`, just for consistency with the previous lines.

File: DataManagementSystem/Utilities/DMSHelpers.py [hunk 7]
```diff
@@ -186,25 +191,32 @@ def getSiteSEMapping( self ):
...
         grid, shortSite, _country = site.split( '.' )
-        if isinstance( tier, ( int, long ) ) and ( grid != 'LCG' or gConfig.getValue( '/Resources/Sites/%s/%s/MoUTierLevel' % ( grid, site ), 999 ) != tier ):
+        if isinstance( tier, ( int, long ) ) and \
+        ( grid != 'LCG' or
+          gConfig.getValue( '/Resources/Sites/%s/%s/MoUTierLevel' % ( grid, site ), 999 ) != tier ):
           continue
-        if isinstance( tier, ( list, tuple, dict, set ) ) and ( grid != 'LCG' or gConfig.getValue( '/Resources/Sites/%s/%s/MoUTierLevel' % ( grid, site ), 999 ) not in tier ):
+        if isinstance( tier, ( list, tuple, dict, set ) ) and \
+        ( grid != 'LCG' or
+          gConfig.getValue( '/Resources/Sites/%s/%s/MoUTierLevel' % ( grid, site ), 999 ) not in tier ):
```
Comment (fstagni) [inline DataManagementSystem/Utilities/DMSHelpers.py]: Did we live with this bug until now?

File: .pylintrc (hunk 4 above)
Comment (phicharp) [reply .pylintrc]: Oh, this was an attempt to get notification for the bad indents of continuation lines, but didn't work... I can remove it!

File: DataManagementSystem/Utilities/DMSHelpers.py (hunk 7 above)
Comment (chaen) [reply DataManagementSystem/Utilities/DMSHelpers.py]: What bug ? It is strictly the same no ?

Comment (phicharp) [reply DataManagementSystem/Utilities/DMSHelpers.py]: I agree ;-)

Comment (fstagni) [reply DataManagementSystem/Utilities/DMSHelpers.py]: Mmm... what did I see? I don't remember... forget it, sorry for the noise!

File: .pylintrc (hunk 5 above)
Comment (phicharp) [reply .pylintrc]: I am a bit lost as this is the message pylint is asking me to include... and the fact is that pylint i uno longer giving me a warning... 
Conclusion: pylint is not really standard nor stable, right?

File: DataManagementSystem/Client/DataManager.py (hunk 6 above)
Comment (fstagni) [reply DataManagementSystem/Client/DataManager.py]: I meant changing line 103 :)

File: .pylintrc (hunk 5 above)
Comment (phicharp) [reply .pylintrc]: Which version of pylint are you using? 
I use 
```
[macprophc] ~/cmtuser/Dirac_v0r0/DIRAC > python /Library/Python/2.7/site-packages/pylint/lint.py --version
lint.py 1.6.5,
astroid 1.4.9
Python 2.7.10 (default, Oct 23 2015, 19:19:21)
```

Comment (fstagni) [reply .pylintrc]: Same as you:

```
DIRAC)  fstagni@DeskTof  ~/pyDevs/DIRAC   v6r19-fixes12  pylint --version                               
pylint 1.6.5, 
astroid 1.4.9
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
[GCC 5.4.0 20160609]
```

Comment (fstagni) [reply .pylintrc]: Can you issue:
```
pylint --list-msgs | grep multiple_types
```
and see if the message is defined?

File: .pylintrc (hunk 4 above)
Comment (phicharp) [reply .pylintrc]: Removed

File: .pylintrc (hunk 5 above)
Comment (phicharp) [reply .pylintrc]: Fixed to redefined-variable-type

**Comment by phicharp** [comment]
I now have weird error notifications from I guess pyDev about gLogger's methods that are not defined through the import... 
Looking a bit in what happes, it is just that the module names used by Alexandre are not conforming with the convention I use and which is apparently standard: module names must be capitalized... Is it a DIRAC standard as well?

**Comment by phicharp** [comment]
OK, fixed (thanks Chris), is due to a lack of indexing in Eclipse when switching branches (because these modules do not exist in v6r17)... The module names are OK, although not conforming to usual DIRAC standard, but this is not the problem here...

**Comment by phicharp** [comment]
I'll now try and update the PR using autopep8 with tabs as 2 spaces (hopefully works)...
Thus will set [WIP]

**Comment by phicharp** [comment]
... and check what is wrong with tests of course...

**Pull request merged**